pandas/core/dtypes/concat.py

```diff
@@ -560,11 +560,6 @@ def _concat_sparse(to_concat, axis=0, typs=None):
 
     fill_values = [x.fill_value for x in to_concat
                    if isinstance(x, SparseArray)]
-
-    if len(set(fill_values)) > 1:
```

Comment:
I think this was leftover from an early implementation that didn't handle multiple fill values. And I think @jorisvandenbossche noticed this in his review, but I misunderstood.

Regardless, we do allow concating Series[sparse] for different fill_values, we just change them all to have the first one's fill value (in `SparseSeries._concat_same_type`)